pkg/frontend/querymiddleware/remote_read.go

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package querymiddleware
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+
+	"github.com/grafana/mimir/pkg/util"
+)
+
+func ParseRemoteReadRequestWithoutCosumingBody(req *http.Request) (url.Values, error) {
```

Comment:
Have you considered a unit test on this?

Reply:
I want to reach an agreement on the output first.